### PR TITLE
feat(presets): split datatables.net grouping 

### PR DESCRIPTION
**Pull request by MindTooth**
## Changes

Tweak https://github.com/renovatebot/renovate/pull/23411 to be more specific as all datatables.net packages aren't always the same.

Project where all presets works: https://github.com/MindTooth/renovate-datatables.net-test/pulls 

## Context

I've had this in the back of my head for some time since my first pull request.  And I wanted to improve it.

Why all the monorepo rules, are because that the individual packages (`datatables.net-buttons`) are combined with an additional style scheme (`datatables.net-buttons-bs4`)  e.g. Bootstrap 4, Bulma, etc.

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [ ] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [ ] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

**Comment by MindTooth** [comment]
I'm not sure why the coverage dropped. 🤔

File: lib/config/presets/internal/monorepo.ts [hunk 1]
```diff
@@ -299,6 +308,20 @@ const patternGroups = {
...
+  'datatables-net-fixedheader': '^datatables.net-fixedheader',
+  'datatables-net-keytable': '^datatables.net-keytable',
+  'datatables-net-responsive': '^datatables.net-responsive',
+  'datatables-net-rowgroup': '^datatables.net-rowgroup',
+  'datatables-net-rowreorder': '^datatables.net-rowreorder',
+  'datatables-net-scroller': '^datatables.net-scroller',
+  'datatables-net-searchbuilder': '^datatables.net-searchbuilder',
+  'datatables-net-searchpanes': '^datatables.net-searchpanes',
+  'datatables-net-select': '^datatables.net-select',
+  'datatables-net-staterestore': '^datatables.net-staterestore',
```
Comment (viceice) [inline lib/config/presets/internal/monorepo.ts:324]: these should be grouped by source url too

File: lib/config/presets/internal/monorepo.ts [hunk 2]
```diff
@@ -67,6 +67,16 @@ const repoGroups = {
...
   'datadog-browser-sdk': 'https://github.com/DataDog/browser-sdk',
+  'datatables-net': [
+    'https://github.com/DataTables/Dist-DataTables',
+    'https://github.com/DataTables/Dist-DataTables-Bootstrap',
+    'https://github.com/DataTables/Dist-DataTables-Bootstrap5',
+    'https://github.com/DataTables/Dist-DataTables-Bulma',
+    'https://github.com/DataTables/Dist-DataTables-DataTables',
+    'https://github.com/DataTables/Dist-DataTables-Foundation',
+    'https://github.com/DataTables/Dist-DataTables-SemanticUI',
+    'https://github.com/DataTables/Dist-DataTables-jQueryUI',
```
Comment (viceice) [inline lib/config/presets/internal/monorepo.ts:78]: do they all have same version?

**Review by viceice — COMMENTED** [review]
we normally group only when they are published together with same version or they need to be grouped because of peer dependency requirements

File: lib/config/presets/internal/monorepo.ts (hunk 1 above)
Comment (MindTooth) [reply lib/config/presets/internal/monorepo.ts:324]: I did this in an attempt to limit the lines needed.  As these can be pulled in groups using regex.  Or else, we'll end up with (8 * 14 =) 112 lines in total to include all by `sourceUrl`.  In other words, they all have a unique `sourceUrl`.

The reason I used `repoGroup` for the base components is that I could not easily target them by a regex.

File: lib/config/presets/internal/monorepo.ts (hunk 2 above)
Comment (MindTooth) [reply lib/config/presets/internal/monorepo.ts:78]: These all have the same version.  They are the base component of DataTables.  The others are plugins.

**Comment by MindTooth** [comment]
Summary, DataTables are composed of a base component, `datatables.net(-bulma|bs4|dt)?`, where e.g. `-dt` is their own styling.  And individual plugins, `datatables.net-scroller(-bulma|bs4|dt)?`, where `scroller` is a plugin.

* The all have unique `sourceUrl`s.
* They are all individually released together, base component is one, each plugin is another.  Each component/plugin has their own version.  In total, 8 packages share their release version.
* In total targeting `sourceUrl`, is (8 * 14 = ) 112 lines
* Almost always, will the user have at least the base component/plugin and an accompanying style (`(-bulma|bs4|dt)?`).

In summary, targeting `sourceUrl` will create many more lines.  Just say so, and it will be done. 👍🏻

File: lib/config/presets/internal/monorepo.ts (hunk 1 above)
Comment (viceice) [reply lib/config/presets/internal/monorepo.ts:324]: we prefer to use source url, even if those are more lines.

we plan to auto group by source url in future, so we don't need this explicit monorepos by source url.

**Review by viceice — COMMENTED** [review]
see comment 🤗

**Review by viceice — APPROVED** [review]
